src/docking_verification/src/docking_verification.py

```diff
@@ -1,66 +1,65 @@
 #!/usr/bin/env python
+"""Calculates offset of chassis (lidar) from pod center."""
+
 import rospy
-from sensor_msgs.msg import PointCloud2
-import sensor_msgs.point_cloud2 as pc2
 import numpy as np
+
+from sensor_msgs import point_cloud2 as pc2
+from sensor_msgs.msg import PointCloud2
 from std_msgs.msg import Float64
+from StateMachine.msg import StateOut
 
-moving_average = np.zeros(5)
-counter = 0
 
-def docking_offset():
-    pub = rospy.Publisher('dock_offset', Float64, queue_size=10)
-    # rospy.init_node('talker', anonymous=True)
-    rate = rospy.Rate(10) # 10hz
-    while not rospy.is_shutdown():
-        # hello_str = "hello world %s" % rospy.get_time()
-        # rospy.loginfo(hello_str)
-        pub.publish(round(np.mean(moving_average), 4))
-        rate.sleep()
+class DockingVerification():
+    """Check state, publish docking offset from pod."""
+    def __init__(self, average_len=10):
+        self.average_len = average_len
+        self.moving_average = np.zeros(average_len)
+        self.counter = 0
+        self.docking_state = False
 
-def velodyne_points_callback(point_cloud):
-    global counter
-    x = []
-    y = []
-    z = []
-    points = np.zeros(3)
-    for p in pc2.read_points(point_cloud, field_names = ("x", "y", "z"), skip_nans=True):
-        # print (" x : %f  y: %f  z: %f" %(p[0],p[1],p[2]))
-        if abs(p[0]) < 15 and abs(p[1]) < 15 and abs(p[2]) < 0.05:
-            # points = np.append(points, np.array([p[0], p[1], p[2]]), axis = 0)
-            x.append(p[0])
-            y.append(p[1])
-            z.append(p[2])
+        self.publisher = rospy.Publisher('dock_offset', Float64, queue_size=10)
+        self.lidar_sub = rospy.Subscriber('system_status', StateOut,
+                                          self.state_listener)
+        self.lidar_sub = rospy.Subscriber("velodyne_points", PointCloud2,
+                                          self.velodyne_points_callback)
 
-    points = np.zeros((3, len(x)))
-    points[0,:] = np.array(x)
-    points[1,:] = np.array(y)
-    points[1,:] = np.array(z)
+    def velodyne_points_callback(self, point_cloud):
+        """Estimate and publish pod center relative to lidar."""
+        if self.docking_state is False: # no need to calculate
+            return
 
-    # print(len(points[0]))
+        x = y = z = []
+        for p in pc2.read_points(point_cloud, field_names=("x", "y", "z"), skip_nans=True):
+            if abs(p[0]) < 15 and abs(p[1]) < 15 and abs(p[2]) < 0.05:
+                x.append(p[0])
+                y.append(p[1])
+                # z.append(p[2])
 
-    mean_x = np.mean(points[0])
-    mean_y = np.mean(points[1])
-    offset = np.sqrt(mean_x**2 + mean_y**2)
+        points = np.zeros((3, len(x)))
+        points[0, :] = np.array(x)
+        points[1, :] = np.array(y)
+        # points[1, :] = np.array(z)
```

Comment:
Shouldn't to be 2 in the comment and change np array size if this is not being used